src/server/v2.0/handler/icon.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/goharbor/harbor/src/controller/icon"
+	"github.com/goharbor/harbor/src/server/v2.0/models"
+	operation "github.com/goharbor/harbor/src/server/v2.0/restapi/operations/icon"
+)
+
+const (
+	maxAge = 604800 // 1 week
+)
+
+func newIconAPI() *iconAPI {
+	return &iconAPI{
+		ctl: icon.Ctl,
+	}
+}
+
+type iconAPI struct {
+	BaseAPI
+	ctl icon.Controller
+}
+
+func (i *iconAPI) GetIcon(ctx context.Context, params operation.GetIconParams) middleware.Responder {
```

Comment:
if err here, how does UI  handler the page render? use the default icons or let the area as empty？

Reply:
I suggest to use a default icon

Reply:
Is it better to log the error and return the default icon?
Considering though this is an api, no one will really call it other than UI.  
Oh we should document that too...